doc/devdocs/locks.rst

```diff
@@ -0,0 +1,127 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free:
+
+ 1. structure code such that only one lock will need to be acquired at a time
+
+ 2. always acquire shared locks in the same order, as given by the table below
+
+ 3. avoid constructs that expect to need unrestricted recursion
+
+Below are all of the locks that exist in the system and the mechanism in use
+that prevents them from being a potential deadlock.
```

Comment:
"all the locks that exist in the system" are not "a potential deadlock," it's incorrect acquisition or release of them that is a potential deadlock. a lock and its acquisition/release are not the same thing.

say "the mechanism for using them in a way that avoids potential deadlocks"[](url)